watertap/costing/util.py

```diff
@@ -100,6 +100,57 @@ def cost_membrane(blk, membrane_cost, factor_membrane_replacement):
     )
 
 
+def cost_rectifier(blk, ac_dc_conversion_efficiency=0.90):
+    """
+    Method to cost rectifiers required for electrified process units that use DC current. Note that if power is supplied
+    by solar or some other DC power generator, this method should not be used. Assumes the unit_model has an `power`
+    variable or parameter.
+
+    Args:
+        ac_dc_conversion_efficiency - Efficiency of the conversion from AC to DC current
+    """
+
+    # create variables on cost block
+    make_capital_cost_var(blk)
+    blk.ac_dc_conversion_efficiency = pyo.Expression(expr=ac_dc_conversion_efficiency)
+    blk.ac_power = pyo.Var(
+        initialize=100,
+        domain=pyo.NonNegativeReals,
+        units=pyo.units.kW,
+        doc="Unit AC power",
+    )
+
+    # use unit.power variable in conversion with efficiency
+    blk.power_conversion = pyo.Constraint(
+        expr=blk.ac_power * blk.ac_dc_conversion_efficiency
+        == pyo.units.convert(blk.unit_model.power, to_units=pyo.units.kW)
+    )
+
+    # USD_2022 embedded in equation
+    rectifier_cost_coeff = {0: 508.6, 1: 2810}
+    blk.rectifier_cost_coeff = pyo.Var(
+        rectifier_cost_coeff.keys(),
+        initialize=rectifier_cost_coeff,
+        units=pyo.units.dimensionless,
+        doc="Rectifier cost coefficients",
+    )
+
+    # refix variables to appropriate costing parameters
+    for index, var in blk.rectifier_cost_coeff.items():
+        var.fix(rectifier_cost_coeff[index])
+
+    # calculate capital cost
+    blk.capital_cost_constraint = pyo.Constraint(
+        expr=blk.capital_cost
+        == blk.rectifier_cost_coeff[1]
+        + (blk.rectifier_cost_coeff[0] * blk.ac_power)
+        * (pyo.units.USD_2020 * pyo.units.kW**-1)
```

Comment:
Comment above says USD_2022 embedded in eq, but USD_2020 implemented in this constraint. Can you clarify the discrepancy?

Reply:
Last comment on this (I think)--should we be converting to `base_currency` here? If not, then why not?

Reply:
Also units were implemented in the wrong place, correcting that now.

Reply:
I did find they're available up until 2021 in `idaes\core\base\costing_base.py`

Reply:
@adam-a-a In the short term I switched to assuming the `USD_2021` conversion is applicable for the parameters regressed. Then I converted the `base_currency` of the test to `USD_2021` to keep the capital cost value the same for the assertion. I also had to then consider the adjusted the electricity price using the CEPCI's to ensure the cost calculation is still as expected. Will push in a second.